# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/AsmJS.cpp
# Commit: 048c2bf91290
# Full Hash: 048c2bf9129090d91f2111579e130d913c871cdd
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-19 10:01:16
# Regressor Bug: 1561521
# File Overlap Count: 1
# Description:
#   Bug 1561521 - Clean up wasm::ToString and drop wasm::ToCString. r=lth
#   
#   wasm::ToCString is losing relevance as we have more constructed types. This
#   commit removes ToCString and replaces all uses with wasm::ToString.
#   
# ==============================================================================

diff -r a72132acc190 -r 048c2bf91290 js/src/wasm/AsmJS.cpp
--- a/js/src/wasm/AsmJS.cpp	Tue Aug 18 16:59:22 2020 +0000
+++ b/js/src/wasm/AsmJS.cpp	Tue Aug 18 16:59:26 2020 +0000
@@ -5859,7 +5859,7 @@
 
   if (f.returnedType() != type) {
     return f.failf(usepn, "%s incompatible with previous return of type %s",
-                   ToCString(type), ToCString(f.returnedType()));
+                   ToString(type).get(), ToString(f.returnedType()).get());
   }
 
   return true;